tests/benchmark/bloatnet/test_bloatnet.py

```diff
@@ -0,0 +1,246 @@
+        + gas_costs.G_KECCAK_256_WORD * 3  # SHA3 dynamic cost (85 bytes)
+        + gas_costs.G_COLD_ACCOUNT_ACCESS  # Cold BALANCE (2600)
+        + gas_costs.G_BASE  # POP balance
+        + gas_costs.G_WARM_ACCOUNT_ACCESS  # Warm EXTCODESIZE (100)
+        + gas_costs.G_BASE  # POP code size
+        + 20  # Overhead for memory operations and loop control
+    )
+
+    # Calculate how many contracts to access
+    available_gas = gas_benchmark_value - intrinsic_gas - 1000  # Reserve for cleanup
+    contracts_needed = int(available_gas // cost_per_contract)
+
+    # Limit to actually deployed contracts
+    num_contracts = min(contracts_needed, NUM_DEPLOYED_CONTRACTS)
+
+    if contracts_needed > NUM_DEPLOYED_CONTRACTS:
+        import warnings
+
+        warnings.warn(
+            f"Test needs {contracts_needed} contracts for "
+            f"{gas_benchmark_value / 1_000_000:.1f}M gas, "
+            f"but only {NUM_DEPLOYED_CONTRACTS} are deployed. "
+            f"Deploy {contracts_needed} contracts for full test coverage.",
+            stacklevel=2,
+        )
+
+    # Generate attack contract with on-the-fly CREATE2 address calculation
+    attack_code = (
+        # Setup memory for CREATE2 address generation
+        # Memory layout: 0xFF + factory_address(20) + salt(32) + init_code_hash(32)
+        Op.MSTORE(0, FACTORY_ADDRESS)
+        + Op.MSTORE8(32 - 20 - 1, 0xFF)  # Prefix for CREATE2
+        + Op.MSTORE(32, 0)  # Initial salt (start from 0)
+        + Op.MSTORE(64, INIT_CODE_HASH)  # Init code hash
+        # Limit counter for number of contracts to access
+        + Op.PUSH2(num_contracts)
+        # Main attack loop
+        + While(
+            body=(
+                # Generate CREATE2 address: keccak256(0xFF + factory + salt + init_code_hash)
+                Op.SHA3(32 - 20 - 1, 85)  # Hash 85 bytes starting from 0xFF
+                # The address is now on the stack
+                + Op.DUP1  # Duplicate for EXTCODESIZE
+                + Op.BALANCE  # Cold access
+                + Op.POP
+                + Op.EXTCODESIZE  # Warm access
+                + Op.POP
+                # Increment salt for next iteration
+                + Op.MSTORE(32, Op.ADD(Op.MLOAD(32), 1))
+            ),
+            # Continue while we haven't reached the limit
+            condition=Op.DUP1 + Op.PUSH1(1) + Op.SWAP1 + Op.SUB + Op.DUP1 + Op.ISZERO + Op.ISZERO,
+        )
+        + Op.POP  # Clean up counter
+    )
+
+    # Deploy attack contract
+    attack_address = pre.deploy_contract(code=attack_code)
+
+    # Run the attack
+    attack_tx = Transaction(
+        to=attack_address,
+        gas_limit=gas_benchmark_value,
+        sender=pre.fund_eoa(),
+    )
+
+    # Post-state: just verify attack contract exists
+    post = {
+        attack_address: Account(storage={}),
+    }
+
+    blockchain_test(
+        pre=pre,
+        blocks=[Block(txs=[attack_tx])],
+        post=post,
+    )
```

Comment:
The CI will fail sine the gas usage is mismatched, but we already resolve the issue by PR https://github.com/ethereum/execution-spec-tests/pull/2155:

```suggestion
    blockchain_test(
        pre=pre,
        blocks=[Block(txs=[attack_tx])],
        post=post,
        skip_gas_used_validation=True,
    )
```